src/statemachine.rs

```diff
@@ -1273,4 +1296,96 @@ impl StateMachine {
 
         self.transaction = Some(try_or!(transaction, |e| cbc.call(Err(e))));
     }
+
+    // Function to interactively manage a specific token.
+    // Difference to register/sign: These want to do something and don't care
+    // with which token they do it.
+    // This function wants to manipulate a specific token. For this, we first
+    // have to select one and then do something with it, based on what it
+    // supports (Set PIN, Change PIN, Reset, etc.).
+    // Hence, we first go through the discovery-phase, then provide the user
+    // with the AuthenticatorInfo and then let them interactively decide what to do
+    pub fn manage(
+        &mut self,
+        timeout: u64,
+        status: Sender<crate::StatusUpdate>,
+        callback: StateCallback<crate::Result<crate::ResetResult>>,
+    ) {
+        // Abort any prior register/sign calls.
+        self.cancel();
+        let cbc = callback.clone();
+
+        let transaction = Transaction::new(
+            timeout,
+            callback.clone(),
+            status,
+            move |info, selector, status, alive| {
+                let mut dev = match Self::init_and_select(info, &selector, true, alive) {
+                    None => {
+                        return;
+                    }
+                    Some(dev) => dev,
+                };
+
+                info!("Device {:?} selected for interactive management.", dev.id());
+
+                // Sending the user the info about the token
+                let (tx, rx) = channel();
+                send_status(
+                    &status,
+                    crate::StatusUpdate::InteractiveManagement((
+                        tx,
+                        dev.get_device_info(),
+                        dev.get_authenticator_info().cloned(),
+                    )),
+                );
+                loop {
```

Comment:
You could also do something like `while alive() && matches!(rx.try_recv(), Err(TryReceiveError::Empty))`

Reply:
Using `while alive()` makes sense, so we can error out early, in case it was aborted before we hit a recv. But I don't really get the second part. We would loose the actual message, in case it's not `TryReceiveError::Empty`, wouldn't we?

Reply:
Nevermind---I was thinking you could spin in a separate loop before calling `recv`, but I missed your earlier comment about CPU usage. Why can't we just `recv` like we do in `ask_user_for_pin`?

Reply:
Ah, because `ask_user_for_pin` is wrong. Physically removing your security key while the pin prompt is shown does not cancel the transaction.